fxa-auth-db-server/test/backend/db_tests.js

```diff
@@ -1208,6 +1211,36 @@ module.exports = function(config, DB) {
                 t.equal(device.callbackPublicKey, '', 'callbackPublicKey updated')
                 t.equal(device.callbackAuthKey, '', 'callbackAuthKey updated')
 
+                // Make the device a zombie, by giving it a non-existent session token
+                return db.updateDevice(ACCOUNT.uid, deviceId, {
+                  sessionTokenId: zombieSessionTokenId
+                })
+                .catch(function () {
+                  t.fail('updating an existing device should not have failed')
+                })
+              })
+              .then(function () {
+                // Fetch all of the devices for the account
+                return db.accountDevices(ACCOUNT.uid)
+              })
+              .then(function (devices) {
+                t.equal(devices.length, 0, 'devices is empty')
```

Comment:
I thought about that and decided against it because it would also require explaining what a zombie device is and why such a thing may exist in the first place, which seemed out of scope. It is not a concept that is referred to elsewhere in our docs so it felt weird to suddenly introduce it here.